packages/trader/src/App/Components/Elements/VideoPlayer/__tests__/volume-control.spec.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import VolumeControl from '../volume-control';
+
+const mocked_props: React.ComponentProps<typeof VolumeControl> = {
+    onVolumeChange: jest.fn(),
+    volume: 0.5,
+    is_mobile: false,
+    is_muted: false,
+    toggleMute: jest.fn(new_value => (mocked_props.is_muted = new_value)),
+};
+
+const volume_bar = 'dt_volume_bar';
+const volume_bar_filled = 'dt_volume_bar_filled';
+const volume_dot = 'dt_volume_dot';
+
+jest.mock('react-transition-group', () => ({
+    ...jest.requireActual('react-transition-group'),
+    CSSTransition: jest.fn(({ children, ...props }) => {
+        return <div>{props.in ? children : null}</div>;
+    }),
+}));
+
+describe('<VolumeControl />', () => {
+    it('should render the component', () => {
+        const { container } = render(<VolumeControl {...mocked_props} />);
+
+        expect(container).not.toBeEmptyDOMElement();
+    });
+
+    it('should call toggleMute with proper argument if user clicked on volume button in order to mute/unmute', () => {
+        const { rerender } = render(<VolumeControl {...mocked_props} />);
+
+        const volume_button = screen.getByRole('button');
```

Comment:
*just in case this test will break:
When you add `role='button'` to satisfy SonarCloud to some divs, this test might break.
In this case, you'll have to find this exact button by its name which can be the icon inside it:
```suggestion
        const volume_button = screen.getByRole('button', name: 'IcSoundOn');
```
and also mock `Icon` component:
```
jest.mock('@deriv/components', () => ({
    ...jest.requireActual('@deriv/components'),
    Icon: jest.fn(({ icon }: { icon: string }) => <div>{icon}</div>),
}));
```

Reply:
If my test didn't break, should I apply this?